### PR TITLE
Fix GoDAM Tab issues for Core Blocks

### PR DESCRIPTION
**Pull request by subodhr258**
Issue: rtCamp/godam-core#534
This pull request introduces support for a new "GoDAM" tab in the WordPress media library interface, allowing integration with GoDAM media items. It also improves how MIME types are determined for GoDAM media entries, ensuring more accurate handling of different media types.

**GoDAM Tab Integration:**

* Added a new `mediaFramePost` module (`media-frame-post.js`) and registered it as a custom view for `wp.media.view.MediaFrame.Post`, enabling the "GoDAM" tab in the media library UI when a valid API key is present. [[1]](diffhunk://#diff-c96ea6334c321bd29a6e93f99ac0b39070d7d6995c8acd4497cfff409f37bfb0R17) [[2]](diffhunk://#diff-c96ea6334c321bd29a6e93f99ac0b39070d7d6995c8acd4497cfff409f37bfb0R97-R100) [[3]](diffhunk://#diff-bbed8d57dc265cb5b5bd683797f105f2803c387cad1de5a910307e3dd466b46bR1-R118)
* Implemented logic in `media-frame-post.js` to render GoDAM media items, handle selection events, and make API calls to create media entries in WordPress for selected GoDAM videos.

**MIME Type Handling Improvements:**

* Modified `prepare_godam_media_item` in `class-media-library-ajax.php` to compute MIME types based on the media item's job type, rather than always using `application/dash+xml`.
* Added a new helper method `get_mime_type_for_job_type` to determine the appropriate MIME type for stream, audio, image, and other job types, improving compatibility and accuracy.

## Recording
Core block support is added for images and audio (not added for video)

https://github.com/user-attachments/assets/723ff163-549c-4392-8edc-6f80bc940afa

### CODE REVIEW

**Comment by KMchaudhary** [comment]
@subodhr258 
Screenshot 2025-11-03 at 3 06 55 PM

Issue: Getting "Attempt recovery" button after image is added and saved and revisit the same page on block editor.
Solution: Remove the "id" attribute, and class present in  element

**Comment by KMchaudhary** [comment]
Also, if we are not providing "GoDAM tab" support to the `core/video` block then remove the GoDAM tab from video block media selector

https://github.com/user-attachments/assets/c68a755e-2f12-41d6-a0f4-29a057fb56a9

**Comment by subodhr258** [comment]
@KMchaudhary I've tested the `wp_get_attachment_url` replacement fix for Core Video block, it seems to be working fine with no side effects in the GoDAM Video block for GoDAM Tab videos.

I'm getting the attempt recovery option for images from GoDAM Tab, because of the same id issue:
```
Content generated by `save` function:

<figure class="wp-block-image"><img src="https://uneltlqeuj.gdcdn.us/jn3p6udp5c/Screenshot - Nov 3, 2025, 10:12:12 AM.png" alt=""/></figure>

Content retrieved from post body:

<figure class="wp-block-image"><img src="https://uneltlqeuj.gdcdn.us/jn3p6udp5c/Screenshot - Nov 3, 2025, 10:12:12 AM.png" alt="" class="wp-image-jn3p6udp5c"/></figure>
```

**Comment by subodhr258** [comment]
TODO: Fix Image block recovery issue (related to ID being invalid for WordPress)
